build-system/tasks/coverage-map.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs').promises;
+const log = require('fancy-log');
+const puppeteer = require('puppeteer');
+const {dist} = require('./dist');
+const {exec} = require('../common/exec');
+const {startServer, stopServer} = require('./serve');
+
+const coverageJsonName = argv.json || 'out.json';
+const testUrl =
+  argv.url || 'http://localhost:8000/examples/everything.amp.html';
+const outHtml = argv.html || 'out.html';
+
+async function doTest() {
+  log(`Opening browser and navigating to ${testUrl}...`);
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  // Enable JavaScript coverage
+  await page.coverage.startJSCoverage();
+  // Navigate to page
+  await page.goto(testUrl);
+  await page.setViewport({
+    width: 1200,
+    height: 800,
+  });
+  log('Scrolling to the end of the page...');
+  await autoScroll(page);
```

Comment:
We need to wait a bit here before scrolling for the page to load. We might have some events we can listen to, or at least we can wait for something in the DOM that will tell us that Runtime is fully initialized. At the very least, we should wait for two things: (1) `<style amp-runtime>` presence, and (2) that body is shown.

Reply:
Can we check for the first thing using `document.querySelector()`?
I assume the second thing can be detected with `document.querySelector('body')`.

Reply:
Yes. The first one is `document.querySelector('style[amp-runtime]')` or such. The second one I think we just need to wait for `document.body.style.visibility === 'visible'`.

Reply:
And re: two lines above: just `!!document.body` is not enough. We need to check that it's visible. Even better that `document.body.style.visibility` is `getComputedStyle(document.body).visibility`.

Reply:
Done.